### PR TITLE
fixed a few bugs in isAbout and source_variable keys in json sidecar files

### PR DESCRIPTION
**Pull request by dbkeator**

### CODE REVIEW

File: src/nidm/experiment/Utils.py [hunk 1]
```diff
@@ -1792,7 +1792,7 @@ def map_variables_to_terms(
                         column_to_terms[current_tuple]["responseOptions"]["unitCode"],
                     )
 
-                if "isAbout" in json_map[json_key[0]]:
+                if "isAbout" in json_map[json_key[0]].keys():
```
Comment (yarikoptic) [inline src/nidm/experiment/Utils.py:1795]: shouldn't be necessary if that is a `dict`

File: src/nidm/experiment/Utils.py [hunk 2]
```diff
@@ -1870,13 +1870,13 @@ def map_variables_to_terms(
                                 )
 
                         print(
-                            f"isAbout: @id = {column_to_terms[current_tuple]['isAbout']['@id']}, label = {column_to_terms[current_tuple]['isAbout']['label']}"
+                            f"isAbout: @id = {column_to_terms[current_tuple]['isAbout'][0]['@id']}, label = {column_to_terms[current_tuple]['isAbout'][0]['label']}"
```
Comment (yarikoptic) [inline src/nidm/experiment/Utils.py:1873]: we figured that it was a wrong thing to do -- there could be lists and scalars

**Comment by yarikoptic** [comment]
overall - without adding tests demonstrating the issue being fixed, there is high chance of playing ping-pong with fixes going back and forward fixing for some use cases and breaking for the other.

File: src/nidm/experiment/Utils.py (hunk 1 above)
Comment (dbkeator) [reply src/nidm/experiment/Utils.py:1795]: It is a list of dictionaries for multiple entries or single entries.  We were testing on an older version of the json sidecar files.  All isAbout entries in either BIDS-compliant JSON or NIDM-compliant JSON-LD sidecar files use a list of dictionaries now.

File: src/nidm/experiment/Utils.py (hunk 2 above)
Comment (dbkeator) [reply src/nidm/experiment/Utils.py:1873]: We were testing on an older version of the json sidecar files. All isAbout entries in either BIDS-compliant JSON or NIDM-compliant JSON-LD sidecar files use a list of dictionaries now.

**Comment by dbkeator** [comment]
@yarikoptic Turns out I already wrote tests for the two variants of the json sidecar files: https://github.com/incf-nidash/PyNIDM/blob/master/tests/experiment/test_map_vars_to_terms.py .  Will refactor the map_variable_to_terms code as we discussed and update the tests...

**Pull request merged**